### PR TITLE
New modal selector

### PR DESCRIPTION
**Pull request by boriskor**
Implement a basic ModalSelector (related to #1163)

Some features are still missing, e.g. checkboxes mode. I'll open a separate issue for all the missing features 🙂.

### CODE REVIEW

File: stories/Selector/ExampleControlled.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ class ControlledSelector extends Component {
                   <Selector
                     title={this.state.title}
                     id={this.state.id}
-                    subTitle={this.state.hasSubtitle ? this.state.subtitle : ''}
+                    subtitle={this.state.hasSubtitle ? this.state.subtitle : ''}
```
Comment (shlomitc) [inline stories/Selector/ExampleControlled.js]: separate PR

File: src/Checkbox/Checkbox.driver.js [hunk 2]
```diff
@@ -15,7 +20,8 @@ const checkboxDriverFactory = ({element, wrapper, component}) => {
     isDisabled: () => isClassExists(element, 'disabled'),
```
Comment (shlomitc) [inline src/Checkbox/Checkbox.driver.js]: this should test also the label design

File: src/Checkbox/Checkbox.driver.js [hunk 3]
```diff
@@ -15,7 +20,8 @@ const checkboxDriverFactory = ({element, wrapper, component}) => {
...
     isIndeterminate: () => $(element).find('.indeterminate').length === 1,
     hasError: () => isClassExists(element, 'hasError'),
-    getLabel: () => element.textContent,
+    getLabel: () => labelDriver().getLabelText(),
+    getLabelDriver: () => labelDriver(),
```
Comment (shlomitc) [inline src/Checkbox/Checkbox.driver.js]: no need to expose it

File: src/ModalSelectorLayout/README.TESTKIT.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
+# ModalSelectorLayout Testkits
+
+> Loader
+
+## Enzyme/ReactTestUtils TestKit API
+
+| method                   | arguments | returned value                           | description                              |
+| ------------------------ | --------- | ---------------------------------------- | ---------------------------------------- |
+| exists                   | -         | boolean                                  | fulfilled if element in the DOM          |
+| mediumLoaderDriver       | -         | [LoaderDriver](https://wix.github.io/wix-style-react/?selectedKind=Core&selectedStory=Loader&full=0&down=0&left=1&panelRight=0) | Main loader shown while the initial items are loading |
```
Comment (shlomitc) [inline src/ModalSelectorLayout/README.TESTKIT.md]: change name

**Comment by argshook** [comment]
@boriskor hi, i skimmed through but this includes a lot of changes and so it's hard to follow. i would say let's split into different PRs if possible. e.g. changes for `Checkbox`, `Radio`, `Search` and such can all be handled in separate PRs imo.

yeah i know, 10 lines of code = 10 issues, 500 LOCs = "looks good". However, let's be on the safe side :)

**Comment by boriskor** [comment]
Splitted non-related changes to a separate PR 😇

**Comment by boriskor** [comment]
FYI, added the missing left padding to the FooterLayout

**Comment by shlomitc** [comment]
@boriskor 
I created a new branch called `2.0.0` for the next breaking change.
Can you please close this PR and open it again against the branch above?

Thanks
